### ISSUE TITLE
Update Data Loader documentation and examples to use DataLoaderFactory

### ISSUE THREAD

**Issue by nateridderman** [ExpediaGroup/graphql-kotlin#1344]
The documentation and examples currently use a deprecated `DataLoader` constructor from graphql-java. That project's Readme instead uses a factory pattern for constructing `DataLoaders` - https://github.com/graphql-java/java-dataloader#examples

It was changed mid last year: https://github.com/graphql-java/java-dataloader/commit/3904878250a53f7c537353b36aac8a1f97eb3f06

**Comment by gvespucci** [comment]
@nateridderman Are you referring to [this `server/data-loaders.md`](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/website/docs/server/data-loaders.md)? Thanks

**Comment by dariuszkuc** [comment]
@gvespucci yes thats one the places to update - code should also be updated (there are only handful of places -> https://github.com/ExpediaGroup/graphql-kotlin/search?q=DataLoader)

TLDR we need to update from constructor invocation `DataLoader<Foo, Bar>()` to `DataLoaderFactory.newDataLoader<Foo, Bar>()`

**Comment by samuelAndalon** [comment]
we will update the the `KotlinDataLoader` interface to provide the `BatchLoader` and create the `DataLoader` internally

**Comment by dariuszkuc** [comment]
Resolved while adding support for batch instrumentation
